lib/pool.js

```diff
@@ -73,6 +76,12 @@ var pools = {
           cb = domain.bind(cb);
         }
         if(err)  return cb(err, null, function() {/*NOOP*/});
+
+        if(client.poolCount === 0){
```

Comment:
Can we remove this? The pool only initializes a client once, so it actually will only emit the `connection` event the first time a connection is constructed. I think this wont be needed.